easybuild/easyblocks/c/cpmd.py

```diff
@@ -0,0 +1,208 @@
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://vscentrum.be/nl/en),
+# the Hercules foundation (http://www.herculesstichting.be/in_English)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing CPMD, implemented as an easyblock
+
+@author: Benjamin Roberts (Landcare Research NZ Ltd) 
+"""
+
+from distutils.version import LooseVersion
+import fileinput
+import glob
+import os
+import platform
+import re
+import shutil
+import sys
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+import easybuild.tools.environment as env
+from easybuild.tools.modules import get_software_root, get_software_version
+from easybuild.tools.run import run_cmd
+from easybuild.tools.systemtools import get_os_type
+import easybuild.tools.toolchain as toolchain
+
+class EB_CPMD(ConfigureMake):
+    """
+    Support for building CPMD
+    """
+
+    @staticmethod
+    def extra_options():
+        """Custom easyconfig parameters for CPMD."""
+        extra_vars = {
+            'base_configuration': [None, "Base configuration from which to start (file name)", CUSTOM],
+        }
+        return ConfigureMake.extra_options(extra_vars)
+
+    def prepare_step(self):
+        super(EB_CPMD, self).prepare_step()
+
+        # create install directory and make sure it does not get cleaned up again in the install step;
+        # the first configure iteration already puts things in place in the install directory,
+        # so that shouldn't get cleaned up afterwards...
+        self.log.info("Creating install dir %s before starting configure-build-install iterations", self.installdir)
+        self.make_installdir()
+        self.cfg['keeppreviousinstall'] = True
+
+    def configure_step(self, cmd_prefix=''):
+        """
+        Configure step
+        """
+
+        config_file_candidates = []
+        config_file_prefix = os.path.join(self.builddir, "CPMD", "configure")
+
+        # Work out a starting configuration file if one is not supplied in the easyconfig
+        if self.cfg['base_configuration']:
```

Comment:
That is done down at line 119 and following. Unless you think that a check should also be put in place up here, with a note that the configuration file could not be found and has been overridden? I'm wary of that; in my opinion, it's better to fail loudly so that the user knows his preferences couldn't be carried out.